core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/udf/PythonUDFOperator.java

```diff
@@ -0,0 +1,700 @@
+                    if (!e.getMessage().contains("Value at index is null")) {
+                        schemaRoot.clear();
+                        closeClientAndServer();
+                        throw new DataflowException(e.getMessage(), e);
+                    } else {
+                        switch (vector.getField().getFieldType().getType().getTypeID()) {
+                            case Int: texeraField = new IntegerField(null); break;
+                            case FloatingPoint: texeraField = new DoubleField(null); break;
+                            case Date: texeraField = new DateField((String) null); break;
+                            case Struct: texeraField = new DateTimeField((String) null); break;
+                            case List: texeraField = new ListField<Span>(null);
+                            default: break;
+                        }
+                    }
+                }
+                texeraFields.add(texeraField);
+            }
+            tuple = new Tuple(texeraSchema, texeraFields);
+            resultQueue.add(tuple);
+        }
+    }
+
+    private Schema convertToTexeraSchema(org.apache.arrow.vector.types.pojo.Schema arrowSchema) {
+        List<Attribute> texeraAttributes = new ArrayList<>();
+        for (Field f : arrowSchema.getFields()) {
+            String attributeName = f.getName();
+            AttributeType attributeType;
+            ArrowType arrowType = f.getFieldType().getType();
+            switch (arrowType.getTypeID()) {
+                case Int:
+                    attributeType = INTEGER;
+                    break;
+                case FloatingPoint:
+                    attributeType = DOUBLE;
+                    break;
+                case Bool:
+                    attributeType = BOOLEAN;
+                    break;
+                case Utf8:
+                case Null:
+                    attributeType = TEXT;
+                    break;
+                case Date:
+                    attributeType = DATE;
+                    break;
+                case Struct:
+                    // For now only Struct of DateTime
+                    attributeType = DATETIME;
+                    break;
+                case List:
+                    attributeType = LIST;
+                    break;
+                default:
+                    closeClientAndServer();
+                    throw (new DataflowException("Unsupported data type "+
+                            arrowType.getTypeID() +
+                            " when converting back to Texera table."));
+            }
+            texeraAttributes.add(new Attribute(attributeName, attributeType));
+        }
+        return new Schema(texeraAttributes);
+    }
+
+    // For now we're only allowing List<Span>. This can (and should) be generalized in the future.
+    private void convertListOfSpans(ImmutableList<Span> spansList, ListVector listVector, int index, String name) throws JsonProcessingException {
```

Comment:
same comment, try make it a `static` function and remove commented out code

Reply:
OK.

Reply:
Done.